File: design_challenges/002/2019_lsp2_javafile/q3.java
import java.io.File;
import java.io.IOException;
import java.io.FileWriter;

public class q3{
	public static void main(String[] args) throws IOException{

		// char *file = "q3java.txt";
		File file = new File("q3java.txt");

		/****** 두번째 매개변수 ******/
		/****** true : 기존 파일의 내용 이후부터 쓰여짐 ******/
		/****** false : 처음부터 쓰여짐 ******/
		// File *writer = fopen(file, "a"); 또는 "w"
		FileWriter writer = new FileWriter(file, false);
		// fprintf(writer, "2019 OSLAB\n");
		writer.write("2019 OSLAB\n");
		writer.write("Linux System Programming\n");
		// fflush(writer);
		writer.flush();
		System.out.printf("DONE\n");
		// writer != NULL
		if(writer != null)
			writer.close(); // fclose(writer);
		
	}
}

File: design_challenges/002/q3.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
	char *file = "q3java.txt";
	/****** 두번째 매개변수 ******/
	/****** true : 기존 파일의 내용 이후부터 쓰여짐 ******/
	/****** false : 처음부터 쓰여짐 ******/
	FILE *writer = fopen(file, "a"); if(!writer)exit(1);
	fprintf(writer, "2019 OSLAB\n");
	fprintf(writer, "Linux System Programming\n");
	fflush(writer);
	printf("DONE\n");
	if(writer != NULL)
	fclose(writer);
	exit(0);
}
